src/eigenspacedecomposition.jl

```diff
@@ -24,21 +24,11 @@ function row_echelon_form!(A::Matrix{T}) where T <: FiniteFields.GF
     return A, l
 end
 
-function row_echelon_form(A::Matrix{T}) where T <: FiniteFields.GF
+function row_echelon_form(A::AbstractMatrix)
     return row_echelon_form!(deepcopy(A))
 end
 
-#=
-function Base.inv(A::Matrix{T}) where T <: FiniteFields.GF
-    n, m = size(A)
-    @assert n == m 
-    B = hcat(deepcopy(A), Matrix{T}(I, n, n))
-    row_echelon_form!(B)
-    return B[:, n+1:end]
-end
-=#
-
-function right_nullspace(M::Matrix{T}) where T <: FiniteFields.GF
+function right_nullspace(M::AbstractMatrix{T}) where T <: FiniteFields.GF
```

Comment:
does the function rely on `T <: GF`? (I don't see how it does)

Reply:
Well, to return the trivial solution...we could replace it by eltype...

Reply:
Or do you say we should have `T<:Any`?

Reply:
I'd just go with `where T`

Reply:
OK,. That's the same as `T<: Any`.